src/main/java/jenkins/plugins/logstash/persistence/BuildData.java

```diff
@@ -276,7 +276,10 @@ public void updateResult()
         Result result = build.getResult();
         this.result = result == null ? null : result.toString();
       }
-      Action testResultAction = build.getAction(AbstractTestResultAction.class);
+      @SuppressWarnings("deprecation") // We don't need transient actions
+      Action testResultAction = build.getActions().stream()
```

Comment:
why not getActions(Class) ?

Reply:
ok, I guess getActions(Class)  does try to get transient actions and that is the problem
wouldn't it better if the build.getAction(Class) were smart enough to determine that I'm requesting a non-transient action?

Reply:
`build.getAction(Class)` has been around for a long long time, even before `TransientActionFactory` were added https://github.com/jenkinsci/jenkins/commit/f1a751f79dfbb975e4c436fd0967323e1ae7b8c6. It eventually was looking for all actions.. Changing it now could be tricky for all consumers..
Maybe what would make more sense eventually would be a `getPersistentAction` / `getNonTransientAction` method in core..